checker/src/types/regexp.rs

```diff
@@ -0,0 +1,91 @@
+use std::{collections::HashMap, fmt};
+
+use regress::{backends, Flags, Match, Regex};
+
+use crate::BinarySerializable;
+
+#[derive(Debug, Clone)]
+pub struct RegExp {
+	source: String,
+	pub groups: u32,
+	pub named_group_indices: HashMap<String, u16>,
+	re: Regex,
+}
+
+impl RegExp {
+	pub fn new(pattern: &str, flags: Option<&str>) -> Self {
+		let source = if let Some(flags) = flags {
+			format!("/{pattern}/{flags}")
+		} else {
+			format!("/{pattern}/")
+		};
+
+		let mut f = Flags::default();
+
+		if let Some(flags) = flags {
+			for flag in flags.chars() {
+				match flag {
+					'd' => unimplemented!("indices for substring matches are not supported"),
+					'g' => unimplemented!("stateful regex is not supported"),
```

Comment:
I think at the moment this can be ignored rather than panicking.

I think the *eventful* design of Ezno can support this in the future. For example evaluating a stateful RegExp could set a `regexp.#used = true` property. That would be a diagnostic that I don't think any other tool does today!